src/app/shared/store/compliance/compliance.service.ts

```diff
@@ -0,0 +1,80 @@
+import { Injectable } from '@angular/core'
+import { Http } from '@angular/http'
+import { Observable } from 'rxjs/Observable'
+
+import { ErrorService } from '../../../shared/services/error.service'
+import { AppConfig } from '../../../shared/utils/config'
+
+@Injectable()
+export class ComplianceService {
+
+  constructor (private http: Http) {}
+
+  getAll (studyId, keys, timeHoles = true): Observable<any> {
+    // TODO: Change when API is ready
+    return this.http.get(`${PARAMS.API_LOCAL}/mock-compliance.json`)
+      .delay(1000)
+      .map(res => {
+        return res.status === 200
+          ? res.json() || null
+          : null
+      })
+      .map(res => {
+        if (res) {
+          return timeHoles
+            ? this.parseComplianceData(this.parseTimeHoles(res, true), keys, timeHoles)
+            : this.parseComplianceData(res.dataset, keys, timeHoles)
+        } else {
+          return null
+        }
+      })
+      .catch(ErrorService.handleError)
+  }
+
+  private parseComplianceData (dataset, keys, timeHoles) {
+    const dates: Date[] = []
+    const values: { [key: string]: number[] } = keys.reduce(
+      (acc, k) => ({ ...acc, [k.key]: [] }), {}
+    )
+
+    dataset.map(data => {
+      keys.map(k => values[k.key].push(data.sample && data.sample[k.key]))
+      dates.push(
+        timeHoles
+        ? data.date
+        :
+        new Date(data.startDateTime))
+    })
+
+    return { keys, values, dates }
+  }
+
+  private parseTimeHoles (res, multi = false) {
```

Comment:
`private parseComplianceData (dataset, keys, timeHoles)`
`private parseTimeHoles (res, multi = false)`

These are either duplicates or very similar to `parseMultiValueData` and `parseTimeHoles` in https://github.com/RADAR-CNS/RADAR-Dashboard/blob/develop/src/app/components/source-graphs/source-graphs.service.ts#L73

We should abstract these functions to be used in both cases and possibly other cases.